packages/react/src/components/shared.tsx

```diff
@@ -0,0 +1,81 @@
+import type { ExpandedRouteConfig } from "@uploadthing/shared";
+
+export const generatePermittedFileTypes = (config?: ExpandedRouteConfig) => {
+    const fileTypes = config ? Object.keys(config) : [];
+
+    const maxFileCount = config
+        ? Object.values(config).map((v) => v.maxFileCount)
+        : [];
+
+    return { fileTypes, multiple: maxFileCount.some((v) => v && v > 1) };
+};
+
+export const capitalizeStart = (str: string) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+export const INTERNAL_doFormatting = (config?: ExpandedRouteConfig): string => {
+    if (!config) return "";
+
+    const allowedTypes = Object.keys(config) as (keyof ExpandedRouteConfig)[];
+
+    const formattedTypes = allowedTypes.map((f) => {
+        if (f.includes("/")) return `${f.split("/")[1].toUpperCase()} file`;
+        return f === "blob" ? "file" : f;
+    });
+
+    // Format multi-type uploader label as "Supports videos, images and files";
+    if (formattedTypes.length > 1) {
+        const lastType = formattedTypes.pop();
+        return `${formattedTypes.join("s, ")} and ${lastType}s`;
+    }
+
+    // Single type uploader label
+    const key = allowedTypes[0];
+    const formattedKey = formattedTypes[0];
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const { maxFileSize, maxFileCount } = config[key]!;
+
+    if (maxFileCount && maxFileCount > 1) {
+        return `${formattedKey}s up to ${maxFileSize}, max ${maxFileCount}`;
+    } else {
+        return `${formattedKey} (${maxFileSize})`;
+    }
+};
+
+export const allowedContentTextLabelGenerator = (
+    config?: ExpandedRouteConfig,
+): string => {
+    return capitalizeStart(INTERNAL_doFormatting(config));
+};
+
+export const progressHeights: { [key: number]: string } = {
```

Comment:
widths?

Reply:
amounts?

Reply:
Since it maps to width property of `after` pseudo-element, I would say `progressWidths` is more suitable